### PR TITLE
chore(Badge): update tests to new react testing library standards

### LINKED ISSUE #7994: Badge test revamp

### PR DESCRIPTION
**Pull request by samuelatefah**
**What**: Closes #7994 

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/Badge/__tests__/Badge.test.tsx [hunk 1]
```diff
@@ -1,10 +1,46 @@
...
-Object.values([true, false]).forEach(isRead => {
-  test(`${isRead} Badge`, () => {
-    const { asFragment } = render(<Badge isRead={isRead}>{isRead ? 'read' : 'unread'} Badge</Badge>);
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders without children', () => {
+  render(
+      <Badge data-testid="badge" />
+  );
+  expect(screen.getByTestId('badge')).toBeVisible();
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/Badge/__tests__/Badge.test.tsx:9]: Can you adjust this a bit to match the pattern you see in the [Avatar tests](https://github.com/Dominik-Petrik/patternfly-react/blob/21905a3b40f45ce108c77d10d1e62d51082836a5/packages/react-core/src/components/Avatar/__tests__/Avatar.test.tsx#L6-L11)? Just for this first test.

The reasoning behind the linked approach is that it serves as a basic sanity check to make sure something renders, while not requiring that the component spreads props to the returned element.

**Review by wise-king-sullyman — CHANGES_REQUESTED** [review]
This looks great! I just have one small thing I'd like to be changed and it should be good to go!

File: packages/react-core/src/components/Badge/__tests__/Badge.test.tsx (hunk 1 above)
Comment (samuelatefah) [reply packages/react-core/src/components/Badge/__tests__/Badge.test.tsx]: Thank you for the review! This should now be updated!

**Review by wise-king-sullyman — APPROVED** [review]
Looks perfect!

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**